tests/document.js

```diff
@@ -0,0 +1,212 @@
+        .expect({message: 'New document created'}, done);
+    });
+    it('asserts that no duplicates are created', function (done) {
+      request
+        .post('/api/documents')
+        .set('x-access-token', token)
+        .send({
+          title: 'Test Doc',
+          content: 'This should work'
+        })
+        .end(function (err, res) {
+          expect(res.body.message).to.equal('Duplicate entry');
+          done();
+        });
+    });
+
+    it('gets all documents available', function (done) {
+      request
+        .get('/api/documents')
+        .set('x-access-token', token)
+        .end(function (err, res) {
+          expect(res.status).to.equal(200);
+          expect(res.body).to.exist;
+          expect(Array.isArray(res.body)).to.equal(true);
+          documentId = res.body[0]._id;
+          done();
+        });
+    });
+
+    it('gets documents within the limit provided', function (done) {
+      request
+          .get('/api/documents')
+          .set('x-access-token', token)
+          .set('limit', 2)
+          .end(function (err, res) {
+            expect(res.status).to.equal(200);
+            expect(res.body).to.exist;
+            expect(Array.isArray(res.body)).to.equal(true);
+            expect(res.body).to.have.length(2);
+            expect(res.body[0].title).to.equal('Hey');
+            done();
+          });
+    });
+
+    it('gets documents with an offset and a limit', function (done) {
+      request
+          .get('/api/documents')
+          .set('x-access-token', token)
+          .set('limit', 2)
+          .set('skip', 1)
+          .end(function (err, res) {
+            expect(res.status).to.equal(200);
+            expect(res.body).to.exist;
+            expect(Array.isArray(res.body)).to.equal(true);
+            expect(res.body).to.have.length(2);
+            expect(res.body[0].title).to.equal('User');
+            done();
+          });
+    });
+
+    it('gets number of documents specified published on a certain date', function (done) {
+      var today = new Date();
+      var day = today.getDate();
+      var month = today.getMonth() + 1;
+      var year = today.getFullYear();
+      var date =  year + '-0' + month + '-' + day;
+
+      request
+          .get('/api/documents?date=' + date)
+          .set('x-access-token', token)
+          .set('limit', 2)
+          .end(function (err, res) {
+            expect(res.status).to.equal(200);
+            expect(res.body).to.exist;
+            expect(Array.isArray(res.body)).to.equal(true);
+            expect(res.body).to.have.length(2);
+            done();
+          });
+    });
+
+    it('returns message if no documents are found', function (done) {
+      request
+          .get('/api/documents?date=2016-09-13')
+          .set('x-access-token', token)
+          .expect({message: 'No documents for this date'}, done);
+    });
+
+    it('gets documents published by the same role level(admin)', function (done) {
```

Comment:
You should test whether the documents returned are those that are published by a role and accessible to the user.